### PR TITLE
[SPARK-46882][SS][TEST] Replace unnecessary AtomicInteger with int

### PR DESCRIPTION
**Pull request by beliefer**
### What changes were proposed in this pull request?
This PR propose to replace unnecessary `AtomicInteger` with int.

### Why are the changes needed?
The variable `value` of `GetMaxCounter` always guarded by itself. So we can replace the unnecessary `AtomicInteger` with int.

### Does this PR introduce _any_ user-facing change?
'No'.

### How was this patch tested?
GA.

### Was this patch authored or co-authored using generative AI tooling?
'No'.

### CODE REVIEW

**Comment by beliefer** [comment]
cc @mridulm

**Comment by beliefer** [comment]
cc @dongjoon-hyun  @HeartSaVioR

File: streaming/src/test/scala/org/apache/spark/streaming/util/WriteAheadLogSuite.scala [hunk 1]
```diff
@@ -238,14 +237,14 @@ class FileBasedWriteAheadLogSuite
...
       def increment(): Unit = synchronized {
-        val atInstant = value.incrementAndGet()
+        val atInstant = value + 1
         if (atInstant > max) max = atInstant
       }
-      def decrement(): Unit = synchronized { value.decrementAndGet() }
-      def get(): Int = synchronized { value.get() }
+      def decrement(): Unit = synchronized { value = value - 1 }
+      def get(): Int = synchronized { value }
       def getMax(): Int = synchronized { max }
```
Comment (dongjoon-hyun) [inline streaming/src/test/scala/org/apache/spark/streaming/util/WriteAheadLogSuite.scala:248]: Actually, this suite failed in the CI with your PR, @beliefer .
```
[info] *** 1 TEST FAILED ***
[error] Failed: Total 440, Failed 1, Errors 0, Passed 439, Ignored 1
[error] Failed tests:
[error] 	org.apache.spark.streaming.util.FileBasedWriteAheadLogSuite
```

Comment (dongjoon-hyun) [reply streaming/src/test/scala/org/apache/spark/streaming/util/WriteAheadLogSuite.scala:248]: Given the failure, I'm not sure if this test suite change is worthy or not.

Comment (beliefer) [reply streaming/src/test/scala/org/apache/spark/streaming/util/WriteAheadLogSuite.scala:248]: Thank you for your reminder. I will check it.

Comment (srowen) [reply streaming/src/test/scala/org/apache/spark/streaming/util/WriteAheadLogSuite.scala:248]: I can't see why this change would be different. But there is probably some subtle issue with memory barriers or whatever. I don't think it's worth changing.

File: streaming/src/test/scala/org/apache/spark/streaming/util/WriteAheadLogSuite.scala [hunk 2]
```diff
@@ -238,14 +237,14 @@ class FileBasedWriteAheadLogSuite
     val executionContext = ExecutionContext.fromExecutorService(fpool)
 
     class GetMaxCounter {
-      private val value = new AtomicInteger()
+      private var value = 0
       @volatile private var max: Int = 0
```
Comment (mridulm) [inline streaming/src/test/scala/org/apache/spark/streaming/util/WriteAheadLogSuite.scala:241]: You can drop the `@volatile` as well.

**Review by mridulm — COMMENTED** [review]
Looks like the tests are passing now - not sure what was changed to fix it.
Agree with @srowen , I dont see a difference between both.

Given this is test suite, it is ok to drop if think it will make tests flakey (the jira should not be "Major" :-) )

File: streaming/src/test/scala/org/apache/spark/streaming/util/WriteAheadLogSuite.scala (hunk 1 above)
Comment (beliefer) [reply streaming/src/test/scala/org/apache/spark/streaming/util/WriteAheadLogSuite.scala:248]: No. I have investigated. The reason is the earlier change is not correct. It's not the issue with memory barriers.
The earlier change is from
```
val atInstant = value.incrementAndGet()
if (atInstant > max) max = atInstant
```
to
```
val atInstant = value + 1
if (atInstant > max) max = atInstant
```

Comment (beliefer) [reply streaming/src/test/scala/org/apache/spark/streaming/util/WriteAheadLogSuite.scala:248]: Now I fixed it.

**Comment by beliefer** [comment]
@srowen I agree with @mridulm .
Let's change the jira from Major to Minor.

**Comment by beliefer** [comment]
cc @srowen @dongjoon-hyun Although this change is just related to test suite, the cost of AtomicInteger is also higher than int.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Comment by dongjoon-hyun** [comment]
Thank you, @beliefer and all.

**Comment by beliefer** [comment]
@dongjoon-hyun @srowen @mridulm Thank you!

**Pull request closed without merging**